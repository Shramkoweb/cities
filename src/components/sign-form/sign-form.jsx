import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {Operation} from "../../reducer/user/user";

const SignForm = (props) => {
  const {onChangeEmail, onChangePassword, emailValue, passwordValue, onFetchLoginData} = props;

  const handleSubmitForm = (evt) => {
    evt.preventDefault();

    onFetchLoginData({
      email: emailValue,
      password: passwordValue
    });
  };

  return (
    <form
      onSubmit={handleSubmitForm}
      className="login__form form"
      action="#"
      method="post"
    >
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">E-mail</label>
        <input
          onChange={onChangeEmail}
          value={emailValue}
          className="login__input form__input"
          name="email"
          placeholder="Email"
          type="email"
          required
        />
      </div>
      <div className="login__input-wrapper form__input-wrapper">
        <label className="visually-hidden">Password</label>
        <input
          onChange={onChangePassword}
          value={passwordValue}
          className="login__input form__input"
          name="password"
          placeholder="Password"
          type="password"
          required
        />
      </div>
      <button className="login__submit form__submit button" type="submit">Sign in</button>
    </form>
  );
};

const mapDispatchToProps = (dispatch) => ({
  onFetchLoginData: (data) => dispatch(Operation.fetchAuthData(data))
});

SignForm.propTypes = {
  onChangeEmail: PropTypes.func.isRequired,
  onChangePassword: PropTypes.func.isRequired,
  emailValue: PropTypes.string.isRequired,
  passwordValue: PropTypes.string.isRequired,
  onFetchLoginData: PropTypes.func.isRequired
};


export {SignForm};
export default connect(null, mapDispatchToProps)(SignForm);
