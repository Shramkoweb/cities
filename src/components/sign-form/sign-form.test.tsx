import * as React from "react";
import {configure, shallow} from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";

import {SignForm} from './sign-form';

jest.mock(`../../index.tsx`, () => jest.fn().mockReturnValue(null));
jest.mock(`../../constants.js`, () => jest.fn().mockReturnValue(null));

configure({adapter: new Adapter()});

it(`SignForm form renders correctly`, () => {
  const signForm = shallow(<SignForm
    onInputChange={jest.fn()}
    sendAuthData={jest.fn()}
    error={null}
    email={`shramko.web@yahoo.com`}
    password={`12345`}
  />);

  expect(signForm).toMatchSnapshot();
});
