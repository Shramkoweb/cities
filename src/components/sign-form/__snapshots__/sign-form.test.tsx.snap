// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignForm form renders correctly 1`] = `ShallowWrapper {}`;
